kernel/src/table_changes/scan.rs

```diff
@@ -213,8 +213,15 @@ impl TableChangesScan {
         }
     }
 
-    /// Get the predicate [`Expression`] of the scan.
-    fn physical_predicate(&self) -> Option<ExpressionRef> {
+    /// Get a shared reference to the [`Schema`] of the table changes scan.
+    ///
+    /// [`Schema`]: crate::schema::Schema
+    pub fn schema(&self) -> &SchemaRef {
+        &self.logical_schema
+    }
+
+    /// Get the predicate [`ExpressionRef`] of the scan.
+    pub fn physical_predicate(&self) -> Option<ExpressionRef> {
```

Comment:
are we sure we want to expose the physical predicate? this would include column-mapped names etc. right?

Reply:
Yeah it would. Perhaps we want the logical predicate that they passed though the builder?

I did this because [we already have this](https://github.com/delta-io/delta-kernel-rs/blob/main/kernel/src/scan/mod.rs#L354) method as pub in `Scan::physical_predicate`

Reply:
A bit of context: https://github.com/delta-io/delta-kernel-rs/pull/512/ changed the (already public)`Scan::predicate` to `Scan::physical_predicate`. The in-kernel callers send it to the parquet reader to enable row group skipping. So it's no worse than exposing the physical read schema to the parquet reader IMO -- the engine only needs to know the concept exists (as in, don't pass the logical schema/predicate there), but doesn't need to _interpret_ either one. Similar story to expression eval in kernel, actually -- engine knows how to evaluate expressions but doesn't need to understand _why_ a given expression needs to be evaluated, nor what Delta spec details the expression relates to.

Since CDF doesn't need to do any parquet reads, we should only expose the physical predicate if we want to pass it to the json reader. There isn't any equivalent to row group skipping for json readers, so most json readers don't do predicate pushdown. But I suppose that's actually engine's decision so we probably _should_ (allow it to) pass the predicate?

Reply:
> Since CDF doesn't need to do any parquet reads, we should only expose the physical predicate if we want to pass it to the json reader

Isn't the physical predicate useful for the data path when reading parquet data files?

> But I suppose that's actually engine's decision so we probably should (allow it to) pass the predicate

For this, should we eagerly do that, or wait until we actually expose a split between metadata and data path for CDF? Currently it's all done through execute.

Reply:
> Isn't the physical predicate useful for the data path when reading parquet data files?

Good point... I was thinking about meta-skipping which doesn't make sense in CDF context.

> should we eagerly [expose/pass the predicate]?

Your call... If duckdb picked up CDF support it would likely want to keep the datapath separate, for example.

Reply:
Thanks @scovich for the detailed reply! And generally I agree with that idea of making public. I'm curious though, if the `read_json_files` already takes a predicate (and i'm not noticing maybe we should rename that to physical_predicate?) how would the API here on the scan be useful?

Reply:
Look at the examples that currently access `Scan::physical_predicate` method. They pass it to the parquet reader they're using to fetch data.